lib/arduino-upload.coffee

```diff
@@ -139,29 +154,37 @@ module.exports = ArduinoUpload =
 		isArduino = fs.existsSync file
 		return {isArduino, workpath, file, name}
 	_build: (options, callback, onerror, port = false) ->
+		if atom.config.get('arduino-upload.autosave') == 'save' or 'save+reopen'
+			if atom.config.get('arduino-upload.autosave') == 'save+reopen'
+				if serial != null
+					serialWasOpen = true
```

Comment:
`serialWasOpen` is only locally defined and later on used, which would yield undefined or something.....it is better to actually define it outside of this inside local if-scope so that it will be defined later on for sure